src/math/Quaternion.js

```diff
@@ -384,6 +384,18 @@ class Quaternion {
 
 	}
 
+	localToWorld( object ) {
```

Comment:
I personally think that math primitives like `Quaternion` or `Vector3` should not know about stuff like local and world coordinates systems. These terms and thus these methods belong to `Object3D`, IMO.

Reply:
>math primitives like Quaternion or Vector3 should not know about stuff like local and world coordinates systems.

Fortunately, they don't.

Like wise for
```js
vector.fromBufferAttribute( attribute )
box3.setFromObject( object );
bufferGeometry.setFromObject( object );
convexHull.setFromObject( object );
```

Reply:
Well, I've never liked this API style. From an engineering point-of-view, these methods belong elsewhere.

Just want to make clear that it's worthy of discussion if the project keeps following this approach.

Reply:
I think you are beginning with a false premise. The object class is not imported into these classes.

Reply:
Also, as I have said multiple times, it is `vector.applyMatrix( m )`, not `matrix.applyToVector( v )`.

Similarly, `quaternion.worldToLocal( object )`; not `object.worldToLocal( quaternion )`.

In any event, we need to first see if this PR is desired. :-)

Reply:
It's not only about the imports. It's the fact the math methods actually work with classes like `Object3D` in their implementations. 

Anyway, it's not my intention to block your contribution. This PR just seemed a good opportunity to highlight something that bothers me for a while now 😇 .

Reply:
I think this is a nice addition -- I think I'd expect it to be on Object3D, though, next to the vector equivalents:

```js
object.localToWorldRotation( quat );
```

Or maybe the existing `localToWorld` and `worldToLocal` functions could be modified to support taking either a quaternion or vector parameter:

```js
object.localToWorld( quat );
object.localToWorld( vec );
```

Reply:
The "vector equivalents" need to be moved to the vector class, where they should have been added in the first place.
```js
vector.localToWorld( obj1 ).worldToLocal( obj2 );
```
not
```js
obj2.worldToLocal( obj1.localToWorld( vector ) );
```

Reply:
@gkjohnson see https://github.com/mrdoob/three.js/issues/13704#issuecomment-684058704 — there is some feeling here that the vector equivalents are in the wrong place. :)